### PR TITLE
Add Cody AI

### PR DESCRIPTION
**Pull request by kukicado**
- Add Cody AI to Code Generation
- Fix Sourcegraph link.

## Requirements

 * [x] This is Software as a Service not self hosted
 * [x] It has a free tier not just a free trial
 * [x] Pricing information is clearly visible without signup or phone calls
 * [x] The submission mentions what is free
 * [x] The submission is not already present in the list
 * [x] The service has contact details of those running it and a privacy policy

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -503,6 +503,7 @@ This list results from Pull Requests, reviews, ideas, and work done by 1100+ peo
 ## Code Generation
 
   * [Appinvento](https://appinvento.io/) — AppInvento is a free No code app builder. In the automatically generated backend code, users have complete access to the source code and unlimited APIs and routes, allowing for extensive integration. The free plan includes three projects, five tables, and a Google add-on.
+  * [Cody AI](https://sourcegraph.com/cody) - Cody is a coding AI assistant that uses AI and a deep understanding of your codebase to help you write and understand code faster. Cody gives developers a choice of LLMs (including local inference), has support for various IDEs, supports all popular programming languages, and has a free plan. The free plan gives developers 20 chat messages (using Claude 3 Sonnet as the LLM) and 500 autocompletions (using the Starcoder 16b) each month.
```
Comment (ripienaar) [inline README.md:506]: we list companies once only, so please add this to an existing entry for sourcegraph. 

Have to be honest though 500 auto completions is not really useful, looking at how many suggestions Goland makes on a daily basis I'd say this wouldnt go far at all.

Comment (kukicado) [reply README.md:506]: Hey there - so Cody AI is a totally different product from Sourcegraph Code Search, they are in two different categories, so I'm not sure if including Cody to the Code Search section makes a lot of sense.

As for the limits, we based the 500 completions based on our usage stats and found that most users fall well under the 500 completions per month, but I will pass this feedback along to our product team.

Comment (ripienaar) [reply README.md:506]: Yeah, you know S3 is quite different from say managed MySQL.  AWS Still listed in one place only.

You're not different to AWS or so, list once only.

Comment (kukicado) [reply README.md:506]: I'm not trying to be rude or argumentative, just trying to understand. AWS is listed and so are a bunch of different AWS services: Lambda, RDS, EC2, etc. and this is true for multiple companies.

Sourcegraph has two very different products in very different categories: 

- Cody AI - a coding AI assistant that lives in your IDE and provides code completion, chat, and other features to help developers write and understand code faster.
- Sourcegraph Search - code search for searching open source repositories, making large scale code changes, and tracking code change insights across your repo.

Like I said, it's very similar to how Vercel has a Generative AI product w/ v0 and a Vercel entry for their app hosting platform. 

Thanks.

Comment (ripienaar) [reply README.md:506]: The services other companiesl ist under one entry are often very varied as well.

We list them once only, that's the policy.  The list would simply be too big if we split it all.  Works for everyone else, please combine them.

Comment (kukicado) [reply README.md:506]: Cool, removed the initial sourcegraph entry.

**Comment by ripienaar** [comment]
Thanks

**Pull request merged**